--- a/kernel/irq/handle.c
+++ b/kernel/irq/handle.c
@@ -20,6 +20,12 @@
 
 #include "internals.h"
 
+#ifdef CONFIG_RTL8196E_IMEM
+#define __iram_hotpath __attribute__((section(".iram")))
+#else
+#define __iram_hotpath
+#endif
+
 #ifdef CONFIG_GENERIC_IRQ_MULTI_HANDLER
 void (*handle_arch_irq)(struct pt_regs *) __ro_after_init;
 #endif
@@ -174,7 +180,7 @@
 	}
 }
 
-irqreturn_t __handle_irq_event_percpu(struct irq_desc *desc)
+irqreturn_t __iram_hotpath __handle_irq_event_percpu(struct irq_desc *desc)
 {
 	irqreturn_t retval = IRQ_NONE;
 	unsigned int irq = desc->irq_data.irq;
@@ -233,7 +239,7 @@
 	return retval;
 }
 
-irqreturn_t handle_irq_event_percpu(struct irq_desc *desc)
+irqreturn_t __iram_hotpath handle_irq_event_percpu(struct irq_desc *desc)
 {
 	irqreturn_t retval;
 
@@ -246,7 +252,7 @@
 	return retval;
 }
 
-irqreturn_t handle_irq_event(struct irq_desc *desc)
+irqreturn_t __iram_hotpath handle_irq_event(struct irq_desc *desc)
 {
 	irqreturn_t ret;
 
